packages/cli/src/modules/mcp/mcp-oauth-token.service.ts

```diff
@@ -96,9 +97,22 @@ export class McpOAuthTokenService {
 			clientId,
 		);
 
-		await this.refreshTokenRepository.remove(refreshTokenRecord);
+		await this.refreshTokenRepository.manager.transaction(async (transactionManager) => {
+			await transactionManager.remove(refreshTokenRecord);
```

Comment:
This one is also prone to a race condition. We could do this using a single sql `DELETE` to fix this:

```ts
const result = trx.delete({
  token: refreshToken,
  clientId,
  expiresAt: LessThan(Date.now())
})
const numAffected = result.affectedRows ?? 0
if (numAffected < 1) {
  throw new Error("Invalid refresh token")
}
```

and generate it only if this check passes